txscript/sign.go

```diff
@@ -393,9 +393,11 @@ sigLoop:
 			// All multisig addresses should be pubkey addresses
 			// it is an error to call this internal function with
 			// bad input.
-			pkaddr := addr.(*dcrutil.AddressSecpPubKey)
-
-			pubKey := pkaddr.PubKey()
+			pkAddr := addr.(stdaddr.SerializedPubKeyer)
+			pubKey, err := secp256k1.ParsePubKey(pkAddr.SerializedPubKey())
+			if err != nil {
+				fmt.Println(err)
+			}
```

Comment:
I'd rather not panic since it could be triggered by a caller feeding it an address type that is not secp256k1 (e.g. ed25519).  The way all of this signing code is currently structured is that it only signs what it can and ignores what it can't.  Given that, it should just continue here instead of printing an error so that it is more consistent with the existing approach.

Updated.